### PR TITLE
#162949235 Add Like and dislike comments feature

### PR DESCRIPTION
**Pull request by Allan690**
#### What does this PR do?
Adds liking and disliking comments feature to the app
#### Description of Task to be completed?
- Add votes field to the comments model
- Add a view to manage likes and dislikes of comments
#### How should this be manually tested?
- Clone the repo: `git clone https://github.com/andela/ah-the-unsullied.git`
- Checkout to the current branch: `git checkout ft-like-unlike-comments-162949235`
-  Create a virtual environment and install the requirements.
```
- virtualenv <name-of-env>
- source <name-of-env>
- pip install -r requirements.txt
```
- Create a .env file and enter the following information
```
     export DATABASE_URL='postgres://USER:PASSWORD@HOST:PORT/NAME'
     export SECRET_KEY="<your_secret_key_here>"
     export HOST_PASSWORD="<Gmail_Password>"
     export HOST_USER="<Gmail_Username>"
     export SMTP_HOST='smtp.gmail.com'
     export EMAIL_PORT='587'
     export FACEBOOK_KEY='<facebook-key>'
     export FACEBOOK_SECRET='<facebook-secret-key>'
     export TWITTER_KEY='<the-twitter-key>'
     export TWITTER_SECRET_KEY='<twitter-secret-key>'
     export GOOGLE_KEY='<google-key>'
     export GOOGLE_SECRET='<google-secret-key>'
     export ACCESS_TOKEN='<access_token>'
     export SECRET_TOKEN='<secret_token>'
```
- Run the migrations: `./manage.py migrate`
- Run the app and navigate to the url: `/api/article/<slug>/comments/<int:id>/like` to post a like and `/api/article/<slug>/comments/<int:id>/dislike` to post a dislike
#### Any background context you want to provide?
None
#### What are the relevant pivotal tracker stories?
#162949235
#### Screenshots (if appropriate)
screenshot 2019-01-29 at 16 33 22

### CODE REVIEW

File: authors/apps/articles/views/comments.py [hunk 1]
```diff
@@ -137,3 +137,50 @@ def delete(self, request, slug, id):
...
             message, status.HTTP_200_OK)
+
+
+class LikeDislikeCommentsView(ListCreateAPIView):
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+    serializer_class = LikeDislikeSerializer
+    renderer_classes = (LikeArticleJSONRenderer,)
+    vote_type = None
+
+    def post(self, request, slug, id):
```
Comment (indungu) [inline authors/apps/articles/views/comments.py]: `id` is reserved keyword in Python please change this parameter to `comment_id` instead. On the plus side, it's more descriptive true?

**Review by indungu — CHANGES_REQUESTED** [review]
Nice work @Allan690. Please make the following change.

File: authors/apps/articles/views/comments.py (hunk 1 above)
Comment (Allan690) [reply authors/apps/articles/views/comments.py]: Sure..making the change

**Review by indungu — DISMISSED** [review]

File: authors/apps/articles/renderers.py [hunk 2]
```diff
@@ -31,29 +31,26 @@ def render(self, data, media_type=None, renderer_context=None):
                 'comments': data,
                 'comments_count': len(data)
             })
-        return json.dumps({
```
Comment (kwanj-k) [inline authors/apps/articles/renderers.py]: Hey, why did you remove this return in the comment renderer, won't it have effects on comments views?
Just wondering

Comment (Allan690) [reply authors/apps/articles/renderers.py]: It is still there...just view the file

**Review by kwanj-k — APPROVED** [review]

**Review by indungu — APPROVED** [review]
